src/plugins/discover/public/application/apps/main/components/layout/discover_layout.tsx

```diff
@@ -214,28 +231,63 @@ export function Discover({
         width: colSettings.width,
       };
       const newGrid = { ...grid, columns: newColumns };
-      opts.setAppState({ grid: newGrid });
+      stateContainer.setAppState({ grid: newGrid });
     },
-    [opts, state]
+    [stateContainer, state]
   );
 
-  const onEditRuntimeField = () => {
-    if (refreshAppState) {
-      refreshAppState();
-    }
-  };
+  const onEditRuntimeField = useCallback(() => {
+    savedSearchRefetch$.next('reset');
+  }, [savedSearchRefetch$]);
+
+  const contentCentered = resultState === 'uninitialized';
+  const showTimeCol = useMemo(
+    () => !config.get('doc_table:hideTimeColumn', false) && !!indexPattern.timeFieldName,
```

Comment:
We have a constant already named `DOC_HIDE_TIME_COLUMN_SETTING` that I think we should prefer here.

Reply:
Agreed and changed